cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/SystemArguments.java

```diff
@@ -509,6 +526,24 @@ public static Map<String, String> getProfileProperties(Map<String, String> args)
     return properties;
   }
 
+  /**
+   *
+   * Gets a boolean from the given map using the given key
+   * If there is no such key, returns null
+   */
+  private static Boolean getBoolean(Map<String, String> map, String key) {
+    String value = map.get(key);
+    if (value == null) {
+      return null;
+    }
+    try {
+      return Boolean.parseBoolean(value);
+    } catch (IllegalArgumentException e) {
+      LOG.warn("cannot parse boolean string");
+    }
```

Comment:
Is this necessary? When does this happen?

Reply:
Since the runtimeargs are just strings. The user will able to specify values other that 'true/false'